<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>853</width>
    <height>605</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: #242526;</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_6">
    <property name="topMargin">
     <number>15</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="2,0,2,0,2">
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,1,0,0">
        <property name="leftMargin">
         <number>20</number>
        </property>
        <property name="rightMargin">
         <number>20</number>
        </property>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>80</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="zerosButton">
            <property name="styleSheet">
             <string notr="true">QPushButton{
  appearance: button;
  backface-visibility: hidden;
  background-color: #405cf5;
  border-radius: 6px;
  border-width: 0;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
  box-sizing: border-box;
  color: #fff;
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  font-size: 15px;
  height: 35px;
  line-height: 1.15;

  outline: none;
  overflow: hidden;
  padding: 0 5px;
  position: relative;
  text-align: center;
  text-transform: none;
  transform: translateZ(0);
  transition: all .2s,box-shadow .08s ease-in;
  touch-action: manipulation;
  width:100px;
}
QPushButton:hover{
background-color: #283999;
}
QPushButton:pressed{
background-color: #1c2973;
}
</string>
            </property>
            <property name="text">
             <string>Zeros</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="polesButton">
            <property name="styleSheet">
             <string notr="true">QPushButton{
  appearance: button;
  backface-visibility: hidden;
  background-color: #405cf5;
  border-radius: 6px;
  border-width: 0;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
  box-sizing: border-box;
  color: #fff;
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  font-size: 15px;
  height: 35px;
  line-height: 1.15;

  outline: none;
  overflow: hidden;
  padding: 0 5px;
  position: relative;
  text-align: center;
  text-transform: none;
  transform: translateZ(0);
  transition: all .2s,box-shadow .08s ease-in;
  touch-action: manipulation;
  width:100px;
}
QPushButton:hover{
background-color: #283999;
}
QPushButton:pressed{
background-color: #1c2973;
}
</string>
            </property>
            <property name="text">
             <string>Poles</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QCheckBox" name="Conj_pair">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">QCheckBox {
    spacing: 10px; /* Adjust the spacing between the checkbox and label */
color:white;
background: rgba(74, 74, 74, 0);
margin-top: 12px;
}

QCheckBox::indicator {
    width: 15px; /* Adjust the width of the checkbox indicator */
    height: 15px; /* Adjust the height of the checkbox indicator */
border-radius: 5px;
}

QCheckBox::indicator:unchecked {
    border: 2px solid #555; /* Border color of the unchecked checkbox */
    background-color: #fff; /* Background color of the unchecked checkbox */
}

QCheckBox::indicator:unchecked:hover {
    border: 2px solid #777; /* Border color when the unchecked checkbox is hovered */
}

QCheckBox::indicator:checked {
    border: 2px solid #192462; /* Border color of the checked checkbox */
    background-color: #405cf5; /* Background color of the checked checkbox */
}

QCheckBox::indicator:checked:hover {
    border: 2px solid #0C1231; /* Border color when the checked checkbox is hovered */
    background-color: #2C40AB; /* Background color when the checked checkbox is hovered */
}
</string>
            </property>
            <property name="text">
             <string>conjugates</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="font">
           <font>
            <pointsize>-1</pointsize>
            <bold>true</bold>
            <kerning>true</kerning>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color:white;
font-size:15px;
margin: 0 0 ;</string>
          </property>
          <property name="text">
           <string>Z_Plane Circle </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="PlotWidget" name="zPlane" native="true">
          <property name="styleSheet">
           <string notr="true">border-radius: 6px;
background: rgba(74, 74, 74, 0);</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="topMargin">
           <number>10</number>
          </property>
          <item>
           <widget class="QComboBox" name="Clear_selection">
            <property name="styleSheet">
             <string notr="true">.QComboBox {
  width: 35px;
  background-color: #333;
  border: 1px solid #333;  /* Change border color to a dark color that complements the theme */
  border-radius: 5px;
  padding: 5px;
  cursor: pointer;
}

.QComboBox::down-arrow {
  width: 10px;
  height: 10px;
  background-color: #405cf5;  /* Change arrow background color to a dark color */
  border-radius: 5px;
  position: absolute;
  right: -10px;
  bottom: 5px;
  cursor: pointer;
}

.QComboBox::down-arrow:hover {
  background-color: #555;  /* Change arrow background color when hovering */
}

.QComboBox::drop-down {
  width: 40px;
  background-color: white;
  border: 1px solid #333;  /* Change drop-down border color to a dark color */
  border-radius: 5px;
  padding: 3px 3px;
  cursor: pointer;
}

.QComboBox::item {
  height: 20px;
  padding: 5px;
  cursor: pointer;
}

.QComboBox::item:selected {
  background-color: #333;  /* Change selected item background color to a dark color */
}

.QComboBox {
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
}
</string>
            </property>
            <item>
             <property name="text">
              <string>Zeros</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Poles</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Both</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clear_button">
            <property name="styleSheet">
             <string notr="true">QPushButton{
  appearance: button;
  backface-visibility: hidden;
  background-color: #405cf5;
  border-radius: 6px;
  border-width: 0;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
  box-sizing: border-box;
  color: #fff;
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  font-size: 15px;
  height: 35px;
  line-height: 1.15;

  outline: none;
  overflow: hidden;
  padding: 0.3px;
  position: relative;
  text-align: center;
  text-transform: none;
  transform: translateZ(0);
  transition: all .2s,box-shadow .08s ease-in;
  touch-action: manipulation;
  width:100px;
}
QPushButton:hover{
background-color: #283999;
}
QPushButton:pressed{
background-color: #1c2973;
}
</string>
            </property>
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>80</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="styleSheet">
         <string notr="true">/* Line style */

  /* Set the width of the line */
  width: 20px;

  /* Set the height of the line */
  height: 5px;

  /* Set the background color of the line */
  background-color:rgb(77, 86, 105);

  /* Set the border of the line */
  border: 10px;

  /* Set the border radius of the line */
  border-radius:5px;




</string>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,4,0,4,0">
        <property name="spacing">
         <number>10</number>
        </property>
        <property name="leftMargin">
         <number>20</number>
        </property>
        <property name="rightMargin">
         <number>20</number>
        </property>
        <property name="bottomMargin">
         <number>10</number>
        </property>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="font">
           <font>
            <pointsize>-1</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color:white;
font-size:15px;
margin: 0 0 ;</string>
          </property>
          <property name="text">
           <string>Magnitude Response</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="PlotWidget" name="magPlot" native="true">
          <property name="styleSheet">
           <string notr="true">border-radius: 6px;
background: rgba(74, 74, 74, 0);</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="font">
           <font>
            <pointsize>-1</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color:white;
font-size:15px;
margin: 0 0 ;</string>
          </property>
          <property name="text">
           <string>Phase Response</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="PlotWidget" name="phasePlot" native="true">
          <property name="styleSheet">
           <string notr="true">border-radius: 6px;
background: rgba(74, 74, 74, 0);</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="correctPhase">
          <property name="styleSheet">
           <string notr="true">QPushButton{
  appearance: button;
  backface-visibility: hidden;
  background-color: #405cf5;
  border-radius: 6px;
  border-width: 0;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
  box-sizing: border-box;
  color: #fff;
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  font-size: 15px;
  height: 35px;
  line-height: 1.15;

  outline: none;
  overflow: hidden;
  padding: 0 5px;
  position: relative;
  text-align: center;
  text-transform: none;
  transform: translateZ(0);
  transition: all .2s,box-shadow .08s ease-in;
  touch-action: manipulation;
  width:100px;
margin: 10px;
}
QPushButton:hover{
background-color: #283999;
}
QPushButton:pressed{
background-color: #1c2973;
}
</string>
          </property>
          <property name="text">
           <string>Correct Phase</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_2">
        <property name="styleSheet">
         <string notr="true">/* Line style */

  /* Set the width of the line */
  width: 20px;

  /* Set the height of the line */
  height: 5px;

  /* Set the background color of the line */
  background-color:rgb(77, 86, 105);

  /* Set the border of the line */
  border: 10px;

  /* Set the border radius of the line */
  border-radius:5px;




</string>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4" stretch="1,0,4,0,4,0,4,0,0">
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>15</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QPushButton" name="importSignal">
            <property name="styleSheet">
             <string notr="true">QPushButton{
  appearance: button;
  backface-visibility: hidden;
  background-color: #405cf5;
  border-radius: 6px;
  border-width: 0;
padding: 6px;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
  box-sizing: border-box;
  color: #fff;
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  font-size: 15px;
  height: 35px;
  line-height: 1.15;

  outline: none;
  overflow: hidden;
  padding: 0 5px;
  position: relative;
  text-align: center;
  text-transform: none;
  transform: translateZ(0);
  transition: all .2s,box-shadow .08s ease-in;
  touch-action: manipulation;
  width:100px;
}
QPushButton:hover{
background-color: #283999;
}
QPushButton:pressed{
background-color: #1c2973;
}
</string>
            </property>
            <property name="text">
             <string>Upload </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="customSignal">
            <property name="styleSheet">
             <string notr="true">QPushButton{
  appearance: button;
  backface-visibility: hidden;
  background-color: #000;
  border: 1px solid gray;
  border-radius: 6px;

padding: 6px;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
  box-sizing: border-box;
  color: #fff;
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  font-size: 15px;
  height: 35px;
  line-height: 1.15;

  outline: none;
  overflow: hidden;
  padding: 0 5px;
  position: relative;
  text-align: center;
  text-transform: none;
  transform: translateZ(0);
  transition: all .2s,box-shadow .08s ease-in;
  touch-action: manipulation;
  width:100px;
}
</string>
            </property>
            <property name="text">
             <string>Custom </string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="font">
           <font>
            <pointsize>-1</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color:white;
font-size:15px;
margin: 0 0 ;</string>
          </property>
          <property name="text">
           <string>Padding Area</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="area" native="true">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(0, 0, 0);</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3"/>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="font">
           <font>
            <pointsize>-1</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color:white;
font-size:15px;
margin: 0 0 ;</string>
          </property>
          <property name="text">
           <string>Input Signal</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="PlotWidget" name="inputSignal" native="true">
          <property name="styleSheet">
           <string notr="true">border-radius: 6px;
background: rgba(74, 74, 74, 0);</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="font">
           <font>
            <pointsize>-1</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color:white;
font-size:15px;
margin: 0 0 ;</string>
          </property>
          <property name="text">
           <string>Output Signal</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="PlotWidget" name="outputSignal" native="true">
          <property name="styleSheet">
           <string notr="true">border-radius: 6px;
background: rgba(74, 74, 74, 0);</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="Speed">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="numOfPoints">
          <property name="styleSheet">
           <string notr="true">color:white;
font-size:10px;
margin: 0 0 ;</string>
          </property>
          <property name="text">
           <string>1 points/sec</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>853</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header location="global">pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
